src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -1055,63 +1056,54 @@ public string Replace(char oldChar, char newChar)
-                }
+            int remainingLength = Length - firstIndex;
+            string result = FastAllocateString(Length);
 
-                if (remainingLength == 0)
-                    return this;
+            int copyLength = firstIndex;
 
-                string result = FastAllocateString(Length);
-
-                fixed (char* pChars = &_firstChar)
-                {
-                    fixed (char* pResult = &result._firstChar)
-                    {
-                        int copyLength = Length - remainingLength;
+            // Copy the characters already proven not to match.
+            if (copyLength > 0)
+            {
+                Buffer.Memmove(ref result._firstChar, ref _firstChar, (uint)copyLength);
+            }
 
-                        // Copy the characters already proven not to match.
-                        if (copyLength > 0)
-                        {
-                            wstrcpy(pResult, pChars, copyLength);
-                        }
+            // Copy the remaining characters, doing the replacement as we go.
+            ref ushort pSrc = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref _firstChar), copyLength);
+            ref ushort pDst = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref result._firstChar), copyLength);
 
-                        // Copy the remaining characters, doing the replacement as we go.
-                        char* pSrc = pChars + copyLength;
-                        char* pDst = pResult + copyLength;
+            if (Vector.IsHardwareAccelerated && remainingLength >= Vector<ushort>.Count)
+            {
+                Vector<ushort> oldChars = new Vector<ushort>(oldChar);
+                Vector<ushort> newChars = new Vector<ushort>(newChar);
 
-                        do
-                        {
-                            char currentChar = *pSrc;
-                            if (currentChar == oldChar)
-                                currentChar = newChar;
-                            *pDst = currentChar;
-
-                            remainingLength--;
-                            pSrc++;
-                            pDst++;
-                        } while (remainingLength > 0);
-                    }
+                do
+                {
+                    Vector<ushort> original = Unsafe.ReadUnaligned<Vector<ushort>>(ref Unsafe.As<ushort, byte>(ref pSrc));
+                    Vector<ushort> equals = Vector.Equals(original, oldChars);
+                    Vector<ushort> results = Vector.ConditionalSelect(equals, newChars, original);
+                    Unsafe.WriteUnaligned(ref Unsafe.As<ushort, byte>(ref pDst), results);
+
+                    pSrc = ref Unsafe.Add(ref pSrc, Vector<ushort>.Count);
+                    pDst = ref Unsafe.Add(ref pDst, Vector<ushort>.Count);
+                    remainingLength -= Vector<ushort>.Count;
                 }
+                while (remainingLength - Vector<ushort>.Count >= 0);
```

Comment:
Can perform the comparison directly without needing a subtraction operation.

```suggestion
                while (remainingLength >= Vector<ushort>.Count);
```